libparsers/tests/xml/dynamic_lighting.cpp

```diff
@@ -0,0 +1,93 @@
+#include "parsers/xml/dynamic_lighting.hpp"
+#include <doctest.h>
+#include <iostream>
+
+using namespace parsers::xml;
+using dynamic_lighting::lighting_info;
+using parsers::errors::error_t;
+using parsers::errors::errors_t;
+
+TEST_SUITE_BEGIN("libparsers - xml - dynamic_lighting");
+
+TEST_CASE("libparsers - xml - dynamic_lighting - test") {
+	std::string const test_value =
+	    "<?xml version=\"1.0\" encoding=\"utf - 8\"?>"
+	    "<openBVE xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsd="
+	    "\"http://www.w3.org/2001/XMLSchema\">"
+	    "<Brightness>"
+	    "<Time>10.00</Time>"
+	    "<AmbientLight>1, 1, 1</AmbientLight>"
+	    "<DirectionalLight>1, 1, 1</DirectionalLight>"
+	    "<LightDirection>1.0, 1.0, 1.0</LightDirection>"
+	    "<CabLighting>80</CabLighting>"
+	    "</Brightness>"
+	    "<Brightness>"
+	    "<Time>12.00</Time>"
+	    "<AmbientLight>160, 160, 160</AmbientLight>"
+	    "<DirectionalLight>160, 160, 160</DirectionalLight>"
+	    "<LightDirection>-0.5, -0.5, -0.5</LightDirection>"
+	    "<CabLighting>255</CabLighting>"
+	    "</Brightness>"
+	    "</openBVE>";
+
+	parsers::errors::multi_error_t output_errors;
+	auto const output = dynamic_lighting::parse("some_file.xml", test_value, output_errors);
+
+	// The if here is needed to check if there's any errors to report
+	// Because otherwise it would print the filename
+	if (!output_errors.empty()) {
+		std::cout << output_errors;
+	}
+	REQUIRE(output_errors.empty());
```

Comment:
This doesn't need to be a REQUIRE, it can be CHECK. Only use REQUIRE if it is guarding against segmentation faults within the tests.